django/db/models/fields/related.py

```diff
@@ -679,6 +680,28 @@ def _check_unique_target(self):
                 ]
         return []
 
+    def _check_conflict_with_managers(self):
+        errors = []
+        for manager in self.opts.managers:
+            for rel_objs in self.model._meta.related_objects:
+                manager_name = manager.name
+                related_object_name = rel_objs.name
+                if manager_name == related_object_name:
+                    field_name = f"{self.model._meta.object_name}.{self.name}"
+                    errors.append(
+                        checks.Error(
+                            f"Related name for '{field_name}' clashes with "
+                            f"manager: '{related_object_name}' name.",
+                            hint=(
+                                "Rename manager name or related_name "
+                                "in conflicted field"
+                            ),
+                            obj=self,
+                            id="fields.E341",
+                        )
+                    )
```

Comment:
We can loop a bit less by collecting the model manager names upfront
I also have a suggestion for the error message and hint (which would need a test update)
```suggestion
        manager_names = {manager.name for manager in self.opts.managers}
        for rel_objs in self.model._meta.related_objects:
            related_object_name = rel_objs.name
            if related_object_name in manager_names:
                field_name = f"{self.model._meta.object_name}.{self.name}"
                errors.append(
                    checks.Error(
                        f"Related name '{related_object_name}' for '{field_name}'"
                        "clashes with the name of a model manager.",
                        hint=(
                            "Rename the model manager or change the related_name "
                            f"argument in the definition for field '{field_name}'."
                        ),
                        obj=self,
                        id="fields.E341",
                    )
                )
```